smll/FaceDetector.cpp

```diff
@@ -285,11 +262,9 @@ namespace smll {
 		if (m_faces.length == 0) {
             // Wait for 5 frames and do face detection
             m_timeout = Config::singleton().get_int(CONFIG_INT_FACE_DETECT_FREQUENCY);
-            m_detectionTimeout = 0;
```

Comment:
I guess this line forces detection if there are no faces found. I still need to test this to confirm.

Reply:
It just doesn't skip frames, 
that's the same as it was before, that was added because the return is removed form detection failing part.

However, forcing detection can be added there immediately

Reply:
I added  a condition that detection happens if detection timeout happened or if there are no faces to track. So this does not have to be set.

Reply:
We could try detection on the same frame where tracking failed. In the previous code detection was performed on the next frame if tracking failed, I just followed that process